frontend/src/component/feature/StrategyTypes/FlexibleStrategy/FlexibleStrategy.tsx

```diff
@@ -41,26 +38,35 @@ const FlexibleStrategy = ({
         updateParameter('rollout', value.toString());
     };
 
-    const resolveStickiness = () =>
-        builtInStickinessOptions.concat(
-            context
-                // @ts-expect-error
-                .filter(c => c.stickiness)
-                .filter(
-                    // @ts-expect-error
-                    c => !builtInStickinessOptions.find(s => s.key === c.name)
-                )
-                // @ts-expect-error
-                .map(c => ({ key: c.name, label: c.name }))
-        );
-
-    const stickinessOptions = resolveStickiness();
-
     const rollout =
         parameters.rollout !== undefined
             ? parseParameterNumber(parameters.rollout)
             : 100;
 
+    const projectStickiness = localStorage.getItem(
+        `defaultStickiness.${projectId}`
+    );
+
+    const resolveStickiness = () => {
+        if (parameters.stickiness === '') {
```

Comment:
I find it difficult to read and multi-nesting doesn't help. Can we flatten the checks somehow and give some intention revealing names?

Reply:
Also with so many ifs I'd be tempted to extract pure fn and test it at the unit level both for confidence and documentation purposes